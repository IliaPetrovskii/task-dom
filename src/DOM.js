/*
  В функцию appendToBody передаются 3 параметра:
  tag - имя тега, content - содержимое тега и count - количество вставок.
  Необходимо, чтобы функция осуществила вставку на страницу указанный тег с указанным содержимым указанное число раз.
  Считаем, что всегда передается тег, допускающий вставку текста в качестве своего содержимого (P, DIV, I и пр.).
*/
export function appendToBody(tag, content, count) {
}

/*
  Создайте дерево вложенных тегов DIV.
  Каждый узел дерева должен содержать childrenCount узлов.
  Глубина дерева задается параметром level.
  Каждый элемент должен иметь класс вида item_n, где n - глубина вложенности элемента. (Нумерацию ведем с единицы).
  Сформированное дерево верните в качестве результата работы функции.
*/
export function generateTree(childrenCount, level) {
}

/*
  Используйте функцию для создания дерева тегов DIV из предыдущего задания.
  Создайте дерево с вложенностью 3 и числом элементов в каждом узле 2.
  Далее замените все узлы второго уровня (т.е. имеющие класс item_2) на теги SECTION.
  Остальную структуру дерева сохраните неизменной, включая классы и те элементы,
  которые находились внутри переписанных тегов.
  Сформированное дерево верните в качестве результата работы функции.
*/
export function replaceNodes() {
}
