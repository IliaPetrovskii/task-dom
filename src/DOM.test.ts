import { appendToBody, generateTree, replaceNodes } from './DOM';

describe('Работа с DOM', () => {
    afterEach(() => {
        document.body.innerHTML = '';
    });

    it('Вставка элементов: 5 последовательных тегов P с содержимым и аттрибутами', () => {
        appendToBody('p', 'Привет, мир!', 5);
        expect(document.body).toMatchSnapshot();
    });

    it('Рекурсивное построение дерева тегов DIV с вложенностью 3, по 3 ветки в узле', () => {
        expect(generateTree(3, 3)).toMatchSnapshot();
    });

    it('Автозамена элементов дерева тегов с сохранением вложенной структуры', () => {
        expect(replaceNodes()).toMatchSnapshot();
    });
});
