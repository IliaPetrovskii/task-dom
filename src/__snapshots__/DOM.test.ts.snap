// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа с DOM Автозамена элементов дерева тегов с сохранением вложенной структуры 1`] = `
<div
  class="item_1"
>
  <section
    class="item_2"
  >
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
  </section>
  <section
    class="item_2"
  >
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
  </section>
</div>
`;

exports[`Работа с DOM Вставка элементов: 5 последовательных тегов P с содержимым и аттрибутами 1`] = `
<body>
  <p>
    Привет, мир!
  </p>
  <p>
    Привет, мир!
  </p>
  <p>
    Привет, мир!
  </p>
  <p>
    Привет, мир!
  </p>
  <p>
    Привет, мир!
  </p>
</body>
`;

exports[`Работа с DOM Рекурсивное построение дерева тегов DIV с вложенностью 3, по 3 ветки в узле 1`] = `
<div
  class="item_1"
>
  <div
    class="item_2"
  >
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
  </div>
  <div
    class="item_2"
  >
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
  </div>
  <div
    class="item_2"
  >
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
    <div
      class="item_3"
    />
  </div>
</div>
`;
